src/js/views/patients/patient/dashboard/dashboard_views.js

```diff
@@ -135,7 +135,11 @@ const ActionItemView = View.extend({
   },
   showDueTime() {
     const isDisabled = this.model.isNew() || !this.model.get('due_date');
-    const dueTimeComponent = new TimeComponent({ time: this.model.get('due_time'), isCompact: true, state: { isDisabled } });
+    const dueTimeComponent = new TimeComponent({
+      time: this.model.get('due_time'),
+      isCompact: true, state: { isDisabled },
+      date: this.model.get('due_date'),
```

Comment:
so.. I think we're going to have to do something like `isOverdue: this.model.isOverdue()` and we'll have to pass that to both the due and time components and then re-render both depending when due_time or due_date change.  Because now the time component is dependent on the whole timestamp, but the date isn't.